apps/server/src/modules/task/uc/task-copy.uc.spec.ts

```diff
@@ -3,15 +3,17 @@ import { Configuration } from '@hpi-schul-cloud/commons';
 import { ObjectId } from '@mikro-orm/mongodb';
 import { ForbiddenException, InternalServerErrorException, NotFoundException } from '@nestjs/common';
 import { Test, TestingModule } from '@nestjs/testing';
-import { BaseDO, User } from '@shared/domain';
 import { CourseRepo, LessonRepo, TaskRepo, UserRepo } from '@shared/repo';
 import { courseFactory, lessonFactory, setupEntities, taskFactory, userFactory } from '@shared/testing';
-import { Action, AuthorizableReferenceType, AuthorizationService } from '@src/modules/authorization';
+import { Action, AuthorizationContextBuilder, AuthorizationService } from '@src/modules/authorization';
 import { CopyElementType, CopyHelperService, CopyStatusEnum } from '@src/modules/copy-helper';
 import { FilesStorageClientAdapterService } from '@src/modules/files-storage-client';
-import { AuthorizableObject } from '@shared/domain/domain-object';
 import { TaskCopyService } from '../service';
 import { TaskCopyUC } from './task-copy.uc';
+import { TaskCopyParentParams } from '../types';
+
+// this.courseRepo.findById(courseId); do not use findOrFail it is possible to pass courseIds in parentParams they do not exists
```

Comment:
Den Kommentar versteh ich nicht. War das ne Notiz für dich oder ist das ein Hinweis für andere?

Reply:
Kurz gesagt da fehlen eine ganze Menge Test. Man kann dort irgendwelche Ids und fake ids rein reichen, das wird alles via as cast ts save gemacht ohne irgendein type guard check. Do not ist hier falsch it use findOrFail. 
Desweiteren ist in den Test das Verhalten für die ganze undefined Behandlung nicht dokumiert. 

Ich wollte hier mit jemanden im Projekt sprechen um zumindest das zu erwartende Verhalten irgendwo zu dokumentieren. Allerdings fühlt sich keiner Verantwortlich. 
Ich werde den Kommentar raus.